src/components/ArtifactDataView.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <section class="artifact-data-view">
+    <component :is="component" :artifact="artifact" />
+  </section>
+</template>
+
+<script lang="ts" setup>
+  import { computed } from 'vue'
+  import ArtifactDataMarkdown from '@/components/ArtifactDataMarkdown.vue'
+  import ArtifactDataResult from '@/components/ArtifactDataResult.vue'
+  import ArtifactDataUnknown from '@/components/ArtifactDataUnknown.vue'
+  import { Artifact } from '@/models'
+
+  const props = defineProps<{
+    artifact: Artifact,
+  }>()
+
+  const component = computed(() => {
+    switch (props.artifact.type) {
+      case 'result':
+        return ArtifactDataResult
+      case 'markdown':
+        return ArtifactDataMarkdown
+      case 'table':
```

Comment:
Probably already considered this but couldn't we use a p-table?

Reply:
Discussed synchronously but short answer yes but we don't have a data structure contract to enforce here yet so this will fallback to the unknown renderer until we do